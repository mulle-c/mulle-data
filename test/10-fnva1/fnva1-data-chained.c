#include <mulle-data/mulle-data.h>
#include <stdio.h>


static void   test_32()
{
   uint32_t   hash;
   char       *s;

   hash = _mulle_fnv1a_init_32();
   hash = _mulle_fnv1a_chained_32( "VfL", 3, hash);
   hash = _mulle_fnv1a_chained_32( " Bochum", 7, hash);
   hash = _mulle_fnv1a_chained_32( " 1848", 5, hash);
   printf( "%#08x\n", hash);
}


static void   test_64()
{
   uint64_t   hash;

   hash = _mulle_fnv1a_init_64();
   hash = _mulle_fnv1a_chained_64( "VfL", 3, hash);
   hash = _mulle_fnv1a_chained_64( " Bochum", 7, hash);
   hash = _mulle_fnv1a_chained_64( " 1848", 5, hash);
   printf( "%#016llx\n", (unsigned long long) hash);
}


int   main( void)
{
   test_32();
   test_64();

   return( 0);
}