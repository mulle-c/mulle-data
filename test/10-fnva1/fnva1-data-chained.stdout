0x2bfecce4
0x8c687db961030a64
